Create pure3.md

jhdalek55 opened the pull request:
This is very much a work in progress and at this point is not even complete. I crafted this from what had been posted in Standard PR #231 and Standard Issue #227. I frankly was not sure what we should do in terms of how we release the information or a patch once a vulnerability is confirmed. 

@hexsecs and others, please add on.

jhdalek55 committed:
Create pure3.md

This is very much a work in progress and at this point is not even complete. I crafted this from what had been posted in Standard PR #231 and Standard Issue #227. I frankly was not sure what we should do in terms of how we release the information or a patch once a vulnerability is confirmed. 

@hexsecs and others, please add on.

mnm678 commented:
Github has a good process for releasing CVEs and patches that allows them to be worked on in secret by a select group of maintainers and contributors, but then made public when they're ready (including comments and other context around the patches). That should work for updating the standard.

The challenge is that if there is a critical vulnerability that would affect implementations, we need a way to know who maintains those implementations in order to communicate the vulnerability before it is made public. With TUF we are able to communicate to all of the open source implementations, but for Uptane we don't have access to source code and so would need some kind of invite-only list of implementers (which sounds very challenging to maintain).

jhdalek55 commented:
Agreed.

pattivacek commented on pure3.md:
```suggestion
Give us a reasonable time to correct the issue before you disclose the vulnerability. We care deeply about security, but we're an open-source project and our team consists mostly of volunteers.
```

pattivacek commented:
I agree, I don't know easy it would be for us to really maintain such a list. The best we could do is have a voluntary signup for early disclosures. Also consider that if we become aware of vulnerabilities in the Standard, we have no way of knowing which implementations might include that vulnerability. Implementers may have intentionally or unintentionally avoided the problem. But I think we still need to make an effort here even if it is a challenge. Best to have the reporting framework in case we need it than to have nothing at all if an emergency comes up.

jhdalek55 committed:
Update pure3.md

Co-authored-by: Patti Vacek <pattivacek@gmail.com>

jhdalek55 commented:
I'm doing some additional work on this PR. Comments and edits are still wanted but understand that I am adding text.

jhdalek55 committed:
adding on

jhdalek55 committed:
additions to PURE3

I fleshed out some proposed procedures. We need reviews to see if what I wrote is feasible. 

I am not sure what would constitute the "Security Analysis" piece for this PURE. Would we need to detail how we would determine the extent of the security risk?

I am assuming that the "backwards compatibility" element would be tied to what the patches might change and not anything to do with the procedure. If so, then backwards compatibility would vary with the vulnerability and its remediation.

jhdalek55 commented:
I've added a proposed procedure. Please review and add on.

jhdalek55 commented:
After our discussion at the 9/13 Standards meeting, it was decided to close this PR. We feel that Uptane should not be in the business of collecting and resolving vulnerabilities. Instead, we need to encourage individuals to report vulnerabilities found in applications using Uptane to whoever released the application. Instead, we are adding information on how to submit errata  for the Standard. See Standard PR #227
